### PR TITLE
Console Logger related fixes.

### PR DESCRIPTION
**Pull request by singhsarab**
- Adding the IsDesignMode boolean in CommandlineOptions
- Make sure Console Logger does not get initialized in design mode.

Upcoming commit
- Add verbosity parameter for ConsoleLogger.

### CODE REVIEW

File: src/Microsoft.TestPlatform.Common/Logging/TestLoggerManager.cs [hunk 1]
```diff
@@ -129,6 +130,19 @@ protected HashSet<string> InitializedLoggers
 
         #region Public Methods
 
+        public void AddLogger(ITestLogger logger, string extensionUri, Dictionary<string, string> parameters)
```
Comment (codito) [inline src/Microsoft.TestPlatform.Common/Logging/TestLoggerManager.cs]: Doc.

File: src/vstest.console/TestPlatformHelpers/TestRequestManager.cs [hunk 2]
```diff
@@ -69,10 +69,11 @@ internal TestRequestManager(CommandLineOptions commandLineOptions, ITestPlatform
...
 
-            // TODO: Is this required for design mode
-            // Add console logger as a listener to logger events.
-            var consoleLogger = new ConsoleLogger();
-            consoleLogger.Initialize(this.testLoggerManager.LoggerEvents, null);
+            if (!this.commandLineOptions.IsDesignMode)
+            {
+                var consoleLogger = new ConsoleLogger();
+                this.testLoggerManager.AddLogger(consoleLogger, ConsoleLogger.ExtensionUri, null);
+             }
```
Comment (codito) [inline src/vstest.console/TestPlatformHelpers/TestRequestManager.cs]: nit: indent

File: test/vstest.console.UnitTests/TestDoubles/DummyLoggerEvents.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
```
Comment (codito) [inline test/vstest.console.UnitTests/TestDoubles/DummyLoggerEvents.cs]: Requires copyright header

File: test/vstest.console.UnitTests/TestDoubles/DummyTestLoggerManager.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿using System;
```
Comment (codito) [inline test/vstest.console.UnitTests/TestDoubles/DummyTestLoggerManager.cs]: Requires copyright header. Usings within namespace.

File: test/vstest.console.UnitTests/TestRequestManagerTests.cs [hunk 5]
```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using vstest.console.UnitTests.TestDoubles;
```
Comment (codito) [inline test/vstest.console.UnitTests/TestRequestManagerTests.cs]: Should go within namespace.

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 6]
```diff
@@ -106,6 +130,26 @@ public void Initialize(TestLoggerEvents events, string testRunDirectory)
             events.TestRunComplete += this.TestRunCompleteHandler;
         }
 
+        public void Initialize(TestLoggerEvents events, Dictionary<string, string> parameters)
```
Comment (codito) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Require tests.

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 7]
```diff
@@ -37,10 +38,33 @@ internal class ConsoleLogger : ITestLogger
...
         public const string FriendlyName = "Console";
 
+        /// <summary>
+        /// Parameter for Verbosity
+        /// </summary>
+        public const string VerbosityParam = "verbosity";
+
         #endregion
 
+        public class Verbosity
```
Comment (codito) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Why not enum?

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 8]
```diff
@@ -37,10 +38,33 @@ internal class ConsoleLogger : ITestLogger
...
+        {
+            public static Verbosity Minimal = new Verbosity("minimal");
+            public static Verbosity Normal = new Verbosity("normal");
+
+            public string level;
```
Comment (codito) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Properties please. Casing.

File: src/vstest.console/Processors/EnableLoggerArgumentProcessor.cs [hunk 9]
```diff
@@ -154,25 +155,32 @@ public void Initialize(string argument)
 
                 if (parseSucceeded)
                 {
-                    // First assume the logger is specified by URI. If that fails try with friendly name.
-                    try
+                    if (loggerIdentifier == "console")
```
Comment (codito) [inline src/vstest.console/Processors/EnableLoggerArgumentProcessor.cs]: ConsoleLogger.FriendlyName?

File: src/vstest.console/Processors/EnableLoggerArgumentProcessor.cs [hunk 10]
```diff
@@ -154,25 +155,32 @@ public void Initialize(string argument)
...
                     {
-                        this.AddLoggerByUri(loggerIdentifier, parameters);
+                        this.loggerManager.AddLogger(new ConsoleLogger(), ConsoleLogger.ExtensionUri, parameters);
                     }
-                    catch (CommandLineException)
+                    else
                     {
-                        string loggerUri;
-                        if (this.loggerManager.TryGetUriFromFriendlyName(loggerIdentifier, out loggerUri))
+                        // First assume the logger is specified by URI. If that fails try with friendly name.
```
Comment (codito) [inline src/vstest.console/Processors/EnableLoggerArgumentProcessor.cs]: This doesn't seem intuitive; user won't specify `/logger:logger://trx/v1`, they will just say `/logger:trx`.

File: src/vstest.console/Processors/EnableLoggerArgumentProcessor.cs [hunk 11]
```diff
@@ -12,6 +12,7 @@ namespace Microsoft.VisualStudio.TestPlatform.CommandLine.Processors
     using Microsoft.VisualStudio.TestPlatform.CommandLine.Processors.Utilities;
 
     using CommandLineResources = Microsoft.VisualStudio.TestPlatform.CommandLine.Resources.Resources;
+    using Microsoft.VisualStudio.TestPlatform.CommandLine.Internal;
```
Comment (codito) [inline src/vstest.console/Processors/EnableLoggerArgumentProcessor.cs]: Require tests.

File: src/vstest.console/Processors/RunTestsArgumentProcessor.cs [hunk 12]
```diff
@@ -148,6 +148,13 @@ public ArgumentProcessorResult Execute()
             Contract.Assert(this.commandLineOptions != null);
             Contract.Assert(!string.IsNullOrWhiteSpace(this.runSettingsManager?.ActiveRunSettings?.SettingsXml));
 
+            if(this.commandLineOptions.IsDesignMode)
```
Comment (codito) [inline src/vstest.console/Processors/RunTestsArgumentProcessor.cs]: nit: space after `if`.

**Review by codito — CHANGES_REQUESTED** [review]
Few suggestions comments. Require unit tests.

**Comment by codito** [comment]
@smadala can you review this please?

File: src/vstest.console/Internal/ConsoleLogger.cs [hunk 13]
```diff
@@ -45,17 +45,10 @@ internal class ConsoleLogger : ITestLoggerWithParameters
 
         #endregion
 
-        public class Verbosity
+        internal enum Verbosity
```
Comment (codito) [inline src/vstest.console/Internal/ConsoleLogger.cs]: Why not private?

File: test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs [hunk 14]
```diff
@@ -137,7 +196,7 @@ public void TestRunCompleteHandlerShouldWriteToConsoleIfTestsPass()
             this.mockOutput.Verify(o => o.WriteLine(string.Format(CultureInfo.CurrentCulture, CommandLineResources.TestRunSummary, 1, 1, 0, 0), OutputLevel.Information), Times.Once());
             this.mockOutput.Verify(o => o.WriteLine(CommandLineResources.TestRunSuccessful, OutputLevel.Information), Times.Once());
         }
-
+
```
Comment (codito) [inline test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs]: nit: remove extra space

File: test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs [hunk 15]
```diff
@@ -115,6 +154,26 @@ public void TestResultHandlerShouldWriteToConsoleIfTestResultEventsAreRaised()
         }
 
         [TestMethod]
+        public void TestResultHandlerShouldWriteToConsoleButSkipPassedTestsForMinimalVerbosity()
+        {
+            var parameters = new Dictionary<string, string>();
+            parameters.Add("verbosity", "minimal");
+            this.consoleLogger.Initialize(new Mock<TestLoggerEvents>().Object, parameters);
+
+            var eventArgs = new T estRunChangedEventArgs(null, this.GetTestResultsObject(), null);
```
Comment (codito) [inline test/vstest.console.UnitTests/Internal/ConsoleLoggerTests.cs]: build failure?

File: test/vstest.console.UnitTests/Processors/EnableLoggersArgumentProcessorTests.cs [hunk 16]
```diff
@@ -73,6 +73,17 @@ public void ExecutorInitializeWithValidArgumentsShouldAddTestLoggerToTestLoggerM
         }
 
         [TestMethod]
+        public void ExecutorInitializeWithValidArgumentsShouldAddConsoleloggerToTestLoggerManager()
```
Comment (codito) [inline test/vstest.console.UnitTests/Processors/EnableLoggersArgumentProcessorTests.cs]: What happens when user provides invalid parameters? e.g. verbosity=quiet

**Review by codito — APPROVED** [review]
Looks good with few suggestions.

File: src/vstest.console/Internal/ConsoleLogger.cs (hunk 13 above)
Comment (singhsarab) [inline src/vstest.console/Internal/ConsoleLogger.cs]: For tests.

File: test/vstest.console.UnitTests/Processors/EnableLoggersArgumentProcessorTests.cs (hunk 16 above)
Comment (singhsarab) [inline test/vstest.console.UnitTests/Processors/EnableLoggersArgumentProcessorTests.cs]: Ignore the params and default to Normal verbosity.

**Pull request merged**